File: Recursion/PrintArray/PrintArray.java
/*
Problem : Print Elements of Array, Straight and reverse
*/

public class PrintArray{

    public static void printArray(int[] arr,int index){

        if(index >= arr.length){
            return;
        }

        System.out.print(arr[index]+" ");
        printArray(arr, index+1);
    }

    public static void printArrayReverse(int[] arr,int index){

        if(index >= arr.length){
            return;
        }

        printArrayReverse(arr, index+1);
        System.out.print(arr[index]+" ");
    }

    public static void main(String[] args) {
        int[] arr = {3,2,4,5,7,21,54,23,562,23};

        System.out.println("Normal Print : ");
        printArray(arr, 0);
        System.out.println();

        System.out.println("Reverse Print : ");
        printArrayReverse(arr, 0);
        System.out.println();
    }
}

File: Recursion/PrintArray/printArray.py
""" 
Problem : Print Elements of Array, Straight and reverse
"""

def printArray(arr,index):

    if index >= len(arr) :
        return
    
    print(arr[index],end=" ")
    printArray(arr,index+1)


def printArrayReverse(arr,index):

    if index >= len(arr) :
        return
    
    printArrayReverse(arr,index+1)
    print(arr[index],end=" ")



arr = [3,2,4,5,7,21,54,23,562,23]
print("Normal print : ",end="")
printArray(arr,0)
print()
print("Reverse Print : ",end="")
printArrayReverse(arr,0)
    